pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -65,22 +65,22 @@ func (r *ReconcileStorageCluster) createStorageClasses(scs []*storagev1.StorageC
 		existing := storagev1.StorageClass{}
 		err := r.client.Get(context.TODO(), types.NamespacedName{Name: sc.Name, Namespace: sc.Namespace}, &existing)
 
-		switch {
-		case err == nil:
-			if existing.DeletionTimestamp != nil {
-				reqLogger.Info(fmt.Sprintf("Unable to restore init object because %s is marked for deletion", existing.Name))
-				return fmt.Errorf("failed to restore initialization object %s because it is marked for deletion", existing.Name)
+		if err == nil || errors.IsNotFound(err) {
+			if err == nil {
+				if existing.DeletionTimestamp != nil {
+					reqLogger.Info(fmt.Sprintf("Unable to restore init object because %s is marked for deletion", existing.Name))
+					return fmt.Errorf("failed to restore initialization object %s because it is marked for deletion", existing.Name)
+				}
```

Comment:
This error message is misleading. We are checking for DeletionTimestamp on StorageClass not Initialization Object.